### ISSUE TITLE
in-app update creates new .desktop file under Linux

### ISSUE THREAD

**Issue by ibahnasy** [telegramdesktop/tdesktop#24716]
### Steps to reproduce

Install Telegram desktop on Linux from the website using the compressed archive.
Perform the in-app upgrade.

### Expected behaviour

No new .desktop files should be created.

### Actual behaviour

A new identical .desktop file has been created automatically.

$ ls -l .local/share/applications/
-rw-rw-r-- 1 user user  699 Jun 24 19:27 appimagekit_6e8d451ac36534d5f16584c78d45570c-Telegram_Desktop.desktop
-rw-rw-r-- 1 user user  699 Jun 26 12:19 appimagekit_829924ee2c988b71f7a32829dc2964df-Telegram_Desktop.desktop

### Operating system

Ubuntu 22.04

### Version of Telegram Desktop

4.0.2

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
Apparently you move the binary everytime? The hash is based on path to the binary.

**Comment by ibahnasy** [comment]
The first time I download Telegram to ~/Downloads and started it, so it created the .desktop file based on that path.
I deleted the .desktop file and moved Telegram to another path and started it, it created .desktop with ~/Downloads again.

So I modified that file manually and never changed Telegram location after that, but after the upgrade, it created the new .desktop file with the exact content as the previous file.
